libsolutil/CommonData.cpp

```diff
@@ -53,6 +53,9 @@ string solidity::util::toHex(uint8_t _data, HexCase _case)
 
 string solidity::util::toHex(bytes const& _data, HexPrefix _prefix, HexCase _case)
 {
+	if (_data.empty())
```

Comment:
What about the prefix?

Reply:
It should check for that one as well. Or maybe even completely remove this check?

Reply:
I actually don't see the point if this early return.

Reply:
Right. Prefix. I totally missed that. But is `0x` even a valid hex value? AFAIK It's not a legal integer literal in C++ or for example Python. Also, I don't think there's any spec for something as simple as hex encoding but Python's standard `binascii.unhexlify()` is one way to decode it I know of and it does not even accept anything with `0x` prefix. I think there should be an assert against this special case unless it's specifically for a use case where we know `0x` on its own would be acceptable.

And looks like you're right that the code would still work without the check.
 `_data.size() - 1` would wrap around to max but it's not used in that case so it's OK.

Reply:
`fromHex` properly works with `"0x"`. Even if `0x` is not a real hex value, it is still better than returning an empty string, because the empty string might just be skipped in a space-separated list.

Reply:
But if it's not a real hex value then maybe an assert would be better to prevent someone from using it unintentionally?

Anyway, this PR is already dead and it probably won't be fixed here so I went ahead and created a new PR with a fix: #9123